tokio/src/io/interest.rs

```diff
@@ -104,6 +128,8 @@ impl Interest {
         match self {
             Interest::READABLE => Ready::READABLE | Ready::READ_CLOSED,
             Interest::WRITABLE => Ready::WRITABLE | Ready::WRITE_CLOSED,
+            #[cfg(any(target_os = "linux", target_os = "android"))]
+            Interest::PRIORITY => Ready::PRIORITY,
```

Comment:
I don't think this branch is covered by the tests right now, it's only used by something like `try_io` from `TcpStream`. It would be great to have a test using `TcpStream::try_io` with the priority interest, for example a test which actually receives the OOB data inside the closure for `try_io`. You can use [`socket2::SockRef::recv_out_of_band`](https://docs.rs/socket2/latest/socket2/struct.SockRef.html#method.recv_out_of_band) for that. Note that you can receive only one byte with that method.

Reply:
Nice, I made the TCP test hit that branch.

Reply:
I think, the `PRIORITY` branch should also cover `Ready::READ_CLOSED`, otherwise an `AsyncFd` with only `Interest::PRIORITY` will not get any notification if the fd closes and maybe wait forever? I don't know.

Reply:
that reasoning makes sense to me, but I don't actually know what the correct choice is here.